Pool_Properties_Design.md

```diff
@@ -198,3 +210,4 @@ None
 
 * This proposal is focusing on storage provision scenario, so all the storage pool properties are based on the current definition of `ProvisionProfile`. If other profiles (such as `ReplicationProfile`, `MigrationProfile`) are designed later, there would be some changes in pool properties.
 * Currently the design of `ReplicationProfile` also contains the properties `RecoveryTimeObjectives`, from my perspective there would no impact to the current design, because they are different use cases for users: one for data provisioning service and the other one is for data replication service. And the reason for adding `RTO` in DataStorageLoS is that user may also need HA when they ask for storage provsioning service, but this feature is definitely optional because not every vendor are not willing to implement this feature.
+* For `FreeCapacity` definition in `StoragePoolSpec`, it's still unclear how to report real free capacity of storage pools, because it varies a lot from different storage drivers. For example, in ceph both the replicated size and thin-provisioning enabled would have an impact on the real capacity. So we should have a deeper discussion about how to normalize this field.
```

Comment:
Thanks @xing-yang , I will have a look. But for now I think it's not the main motivation about this proposal, so I suggest we finish the pool capability properties design if there is no quesion, and I will submit another patch to fix it. Is that ok for you? @jungleboyj @stmcginnis

Reply:
I am a little concerned not having that clearly defined. It is something that we will need to make sure is explicit to avoid confusion down the line. But I do think it would be OK to agree on this overall spec for now as long as we have a plan to go back and clarify this point afterwards.

Reply:
Yes, we should have this clearly defined.  The over subscription in thin provisioning support in Cinder was added in the Kilo release.  Before that, Cinder only had a free capacity.  It takes years to build a project as mature as Cinder.  We still have a lot to catch up.

Reply:
Leon,

Sorry I hadn't had a chance to respond, but yes, definition of the FreeCapacity field was the concern I was raising.  It has also been a struggle for Cinder to come to agreement on how to handle the concept.  We recently created this document to try to help with some of the issues:  https://specs.openstack.org/openstack/cinder-specs/specs/queens/provisioning-improvements.html
We defined free capacity there as follows:
It is the current physical capacity available in the storage array’s pool being used by Cinder. The number and volume sizes of the thin and thick volumes that have been provisioned by Cinder or directly in the storage array are irrelevant here.

This is currently being reported by the drivers as free_capacity_gb and, as the name indicates, should be reported in GB and with a precision no greater than 2 decimals.

If the storage array has 5TB of space with a total of 3TB available for all its pools but Cinder is using a pool that has a limit of 1TB of which it has already used 400GB and someone has manually created volumes outside of Cinder that are currently using 124GB of space, then the driver should be reporting a free_capacity_gb of 500GB (1TB = 1024GB = 400GB + 124GB + 500GB).

Reply:
Thanks @jungleboyj .  I remember Gorka has made some clarifications to the capacity definition but couldn't locate this spec:).  That will give us a whole picture of what Cinder has and we can start from here.